### ISSUE TITLE
Warning: Failed prop type: The prop `name` is marked as required in `Field`, but its value is `undefined`.

### ISSUE THREAD

**Issue by firepol** [marmelab/react-admin#3434]
I have an react-admin panel to manage accounts, balances and currencies...

DB Model (summed up):
- account (id, name etc) has balances
- each balance (id, total) refers to a currency (currency_id) and to a parent account (account_id)
- currency: id, name, slug, symbol

**What you were expecting:**

The account dropdown works. So I expect the currency dropdown also to work and show all my currencies.

**What happened instead:**

Currency dropdown: empty (but if I click it, I get redirected to the currency of the correct id, so currency_id works)

![image](https://user-images.githubusercontent.com/1702718/61591723-59cf1d80-abca-11e9-88cb-fa266a80053f.png)

- Account list works (/api/accounts)
- Account by id (/api/accounts?id=1) works
- Currency list works (/api/currencies)
- Currency by id (/api/currencies?id=1) works

Edit/Create Balance, the reference to the account works as expected using this code:
```
      <ReferenceInput source="account_id" reference="accounts">
        <SelectInput optionText="name" />
      </ReferenceInput>
```

The reference to the currency does not work, the code is basically the same as the one used for account (except it's `currency` instead of `account`), but it doesn't work and I get this error:

```
      <ReferenceField source="currency_id" reference="currencies">
        <SelectInput optionText="name" />
      </ReferenceField>
```

Note: the currency list contains 2201 elements, can it be that these are too many for the SelectInput?

I tried to use also the AutoComplete control... Same error.

The response for the /api/currencies?id=1 is the following (similar to the ones for accounts?id=):

```
[
  {
    "id": 1, 
    "name": "US Dollar", 
    "slug": "usd", 
    "symbol": "USD"
  }
]
```

**Steps to reproduce:**
I don't know how to reproduce this... just for one resource (accounts) it works, for another (currencies) it doesn't.

If you know a way to debug this, maybe I can retrieve more information and fix it myself, but I'm new to react and to react-admin, so I'm stuck here. Eventually I can open source my code and see.

**Code snippet**

See here: https://github.com/firepol/crypto-accounting-admin/blob/master/src/balances/balances.js
(you can also check the rest of the repo to see what kind of data provider I'm using etc.)

**Environment**

* React-admin version: 2.9.5
* React version:
* Browser: Chromium
* Stack trace (in case of a JS error):

![image](https://user-images.githubusercontent.com/1702718/61591776-d235de80-abca-11e9-9690-3d600ea90c0f.png)

Note that in the edit page, the call to accounts, indicated with a red arrow- (options) is followed by `api/accounts?_end=25&_order=DESC&_sort=id&_start=0` (again options), followed by 2 calls to the relative "GET" method (and with the proper response json)

I don't understand why there isn't another call to currencies to fetch also the entire list (with these querystrings: `_end=25&_order=DESC&_sort=id&_start=0`). See arrows in blue in the screenshot.

```
index.js:1375 Warning: Failed prop type: The prop `name` is marked as required in `Field`, but its value is `undefined`.
    in Field (created by FormFieldView)
    in FormFieldView (created by DefaultValueView)
    in DefaultValueView (created by Connect(DefaultValueView))
    in Connect(DefaultValueView) (created by WithFormField)
    in WithFormField (at balances.js:115)
    in a (created by Link)
    in Link (created by Link)
    in Link (created by WithStyles(Link))
    in WithStyles(Link) (created by ReferenceFieldView)
    in ReferenceFieldView (created by UnconnectedReferenceFieldController)
    in UnconnectedReferenceFieldController (created by Connect(UnconnectedReferenceFieldController))
    in Connect(UnconnectedReferenceFieldController) (created by ReferenceField)
    in ReferenceField (created by EnhancedReferenceField)
    in EnhancedReferenceField (at balances.js:114)
    in div (created by Labeled)
    in div (created by FormControl)
    in FormControl (created by WithStyles(FormControl))
    in WithStyles(FormControl) (created by Labeled)
    in Labeled (created by WithStyles(Labeled))
    in WithStyles(Labeled) (created by FormInput)
    in div (created by FormInput)
    in FormInput (created by WithStyles(FormInput))
    in WithStyles(FormInput) (created by SimpleForm)
    in div (created by CardContent)
    in CardContent (created by WithStyles(CardContent))
    in WithStyles(CardContent) (created by CardContentInner)
    in CardContentInner (created by WithStyles(CardContentInner))
    in WithStyles(CardContentInner) (created by SimpleForm)
    in form (created by SimpleForm)
    in SimpleForm (created by Form(SimpleForm))
    in Form(SimpleForm) (created by Connect(Form(SimpleForm)))
    in Connect(Form(SimpleForm)) (created by ReduxForm)
    in ReduxForm (created by Context.Consumer)
    in translate(ReduxForm) (created by Connect(translate(ReduxForm)))
    in Connect(translate(ReduxForm)) (at balances.js:108)
    in div (created by Paper)
    in Paper (created by WithStyles(Paper))
    in WithStyles(Paper) (created by Card)
    in Card (created by WithStyles(Card))
    in WithStyles(Card)
    in div
    in Unknown (created by WithStyles(Component))
    in WithStyles(Component) (created by UnconnectedEditController)
    in UnconnectedEditController (created by Context.Consumer)
    in translate(UnconnectedEditController) (created by Connect(translate(UnconnectedEditController)))
    in Connect(translate(UnconnectedEditController))
    in Unknown (created by Edit)
    in Edit (at balances.js:107)
    in BalanceEdit (created by WithPermissions)
    in WithPermissions (created by Connect(WithPermissions))
    in Connect(WithPermissions) (created by getContext(Connect(WithPermissions)))
    in getContext(Connect(WithPermissions)) (created by Route)
    in Route (created by Resource)
    in Switch (created by Resource)
    in Resource (created by Connect(Resource))
    in Connect(Resource) (at App.js:14)
    in Route (created by RoutesWithLayout)
    in Switch (created by RoutesWithLayout)
    in RoutesWithLayout (created by Route)
    in div (created by Layout)
    in main (created by Layout)
    in div (created by Layout)
    in div (created by Layout)
    in Layout (created by WithStyles(Layout))
    in WithStyles(Layout) (created by Route)
    in Route (created by withRouter(WithStyles(Layout)))
    in withRouter(WithStyles(Layout)) (created by Connect(withRouter(WithStyles(Layout))))
    in Connect(withRouter(WithStyles(Layout))) (created by LayoutWithTheme)
    in MuiThemeProvider (created by LayoutWithTheme)
    in LayoutWithTheme (created by Route)
    in Route (created by CoreAdminRouter)
    in Switch (created by CoreAdminRouter)
    in div (created by CoreAdminRouter)
    in CoreAdminRouter (created by Connect(CoreAdminRouter))
    in Connect(CoreAdminRouter) (created by getContext(Connect(CoreAdminRouter)))
    in getContext(Connect(CoreAdminRouter)) (created by Route)
    in Route (created by CoreAdminBase)
    in Switch (created by CoreAdminBase)
    in Router (created by ConnectedRouter)
    in ConnectedRouter (created by CoreAdminBase)
    in TranslationProviderView (created by Connect(TranslationProviderView))
    in Connect(TranslationProviderView) (created by CoreAdminBase)
    in Provider (created by CoreAdminBase)
    in CoreAdminBase (created by withContext(CoreAdminBase))
    in withContext(CoreAdminBase) (at App.js:12)
    in App (at src/index.js:7)
```

**Comment by Kmaschta** [comment]
The ReferenceField expects that the currency request returns an object, but if your request `/api/currencies?id=1` returns an array and that your data provider isn't adapted, the reference shouldn't work properly.

You can see the related documentation : https://marmelab.com/react-admin/DataProviders.html#writing-your-own-data-provider

This seems to be a usage problem, and not an error coming from React Admin.
I can only recommend you to ask this question on [Stack OverFlow](http://stackoverflow.com/questions/tagged/react-admin).

Unlike GitHub, Stack Overflow has great SEO, gamification, voting, and reputation. That's why we chose it, and decided to keep GitHub issues only for bugs and feature requests.

So I'm closing this issue, and inviting you to ask your question at:

http://stackoverflow.com/questions/tagged/react-admin

Once you get a response, please continue to hang out on the react-admin channel in Stack Overflow. That way, you can help newcomers and share your expertise!

Feel free to re-open if you can reproduce an error coming from React Admin.

**Comment by firepol** [comment]
@Kmaschta please check the behavior of react-admin. In another list I have a reference to accounts, and in that list I see many calls like this one:

`http://127.0.0.1:5000/api/accounts?id=4&id=5&id=6&id=7&id=8&id=9&id=10&id=13&id=14&id=15&id=17&id=18&id=22`

Such a call `/api/accounts?id=1&id=2` expects an array of objects, not just one object.

Not to be confused with a call to `/api/accounts/2`: this one should return one object. This one is e.g. when I click in my list an account, to go to the account edit page.

So basically in the backend I had to implement 2 endpoints.

One for `/api/accounts/<id>`
Another one for `/api/accounts/<filter>`: I had to modify my query to allow to filter accounts by ids (e.g. `WHERE ID in [list of ids]`.

That said, ok I will try my luck on stackoverflow. But to me this doesn't look like a usage issue, it's a bug in react-admin.

**Comment by Kmaschta** [comment]
Fair enough! That being said, maintainers cannot browse all react admin users codebase in order to find bugs.

The best way for us to track down an issue is to have a minimal example (preferably within a CodeSandbox, a gist, or a standalone repository) that can reproduce the issue.

If you're able to provide us something small and reproducible, I'll be glad to re-open this issue!
